app/controllers/users_controller.rb

```diff
@@ -35,21 +35,18 @@ def update
   end
 
   def promote
-    @users = User.where.not(staff: true).sort_by(&:proper_name)
+    users = User.where.not(staff: true)
+    @users = []
+    users.each do |user|
```

Comment:
You can do this more simply (without maintaining two copies of the same data structure) using map.